PolyEngine/Engine/Src/HeavyTaskImplementation.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include "HeavyTaskBase.hpp"
+#include "World.hpp"
+
+namespace HeavyTaskHelper
+{
+	template <std::size_t... T>
+	struct index {};
+
+	template <std::size_t N, std::size_t... T>
+	struct gen_seq : gen_seq<N - 1, N - 1, T...> {};
+
+	template <std::size_t... T>
+	struct gen_seq<0, T...> : index<T...> {};
+}
+
+namespace Poly
+{
+	class DestroyEntityHeavyTask : public HeavyTaskBase
+	{
+	private:
+		const UniqueID *Id;
+	public:
+		DestroyEntityHeavyTask(const UniqueID &entityID)
+		{
+			Id = new UniqueID(entityID);
+		}
+		~DestroyEntityHeavyTask()
+		{
+			if (Id) delete Id;
+		}
+
+		virtual void Execute(World* w)
+		{
+			w->DestroyEntity(*Id);
+		}
+
+		virtual String GetDescription() { return String("Destroy entity"); }
+	};
+	//---------------------------------------------------------------
+	template<typename T>
+	class RemoveComponentHeavyTask : public HeavyTaskBase
+	{
+	private:
+		const UniqueID *Id;
+	public:
+		RemoveComponentHeavyTask(const UniqueID &entityID)
+		{
+			Id = new UniqueID(entityID);
+		}
+		~RemoveComponentHeavyTask()
+		{
+			if (Id) delete Id;
```

Comment:
If the above comment is applied then destructor is not necessary so please remove it.